### ISSUE TITLE
IPV6 can't be compiled as a module

### ISSUE THREAD

**Issue by nekromant** [linux-sunxi/linux-sunxi#109]
Selecting to compile IPV6 as a module results in the following link errors.  (this is current sunxi-3.4 branch)

```
LD      .tmp_vmlinux1
net/built-in.o: In function `tcp_nuke_addr':
sysctl_net.c:(.text+0x47660): undefined reference to `rt6_lookup'
sysctl_net.c:(.text+0x476c0): undefined reference to `in6addr_any'
make[5]: *** [.tmp_vmlinux1] Error 1
```

This is bad, since it breaks things in OpenWRT, which I'm porting to sunxi right now.

**Comment by SharkyRawr** [comment]
+1

**Comment by nvdl** [comment]
+1

**Comment by amery** [comment]
tried making a `modprobe.conf` alias to fool openwrt?

**Comment by nekromant** [comment]
Nope, not sure how it would help. Compiling as built-in works, but due to how the openwrt buildsystem works, if kmod-ipv6 is selected - it will override  the .config value during kernel build. So far the only solution for me was to hack openwrt kconfig files, but that doesn't sound good. 
P.S. Any chance the 3.7 branch will be cleaner, so that we can just use git format-patches and drop them to patches-3.7/ of the openwrt tree? Due to the current 3.4 state the only way to go or OpenWRT is using an external kernel tree, but that breaks some things, since for OpenWRT there are a few extra patches to be applied on top of it during build.

**Comment by amery** [comment]
there won't be a 3.7. after 3.4 next is mainline.

**Comment by SharkyRawr** [comment]
I have been trying to cross compile the default/unmodified (no openwrt or other platform involved) stage/sunxi-3.4 branch for my Hackberry (A10) and get the same error with ipv6 as a module. The kernel won't even link properly because of it. Works fine "compiled-in" though.

**Comment by amery** [comment]
@schumann2k: yes, it's known. part of the android legacy of this tree. ipv6 as module is not supported.

**Comment by SharkyRawr** [comment]
I see, I've been wondering so thanks for the explanation. :)
